packages/flame/lib/src/collisions/quadtree/broadphase.dart

```diff
@@ -0,0 +1,164 @@
+    int maxObjects = 25,
+    int maxLevels = 10,
+  }) : tree = QuadTree<T>(
+          mainBoxSize: mainBoxSize,
+          maxObjects: maxObjects,
+          maxLevels: maxLevels,
+        );
+
+  final QuadTree tree;
+
+  final activeCollisions = HashSet<T>();
+
+  ExternalBroadphaseCheck broadphaseCheck;
+  ExternalMinDistanceCheck minimumDistanceCheck;
+  final _broadphaseCheckCache = <T, Map<T, bool>>{};
+
+  final _cachedCenters = <ShapeHitbox, Vector2>{};
+
+  final potentials = HashSet<CollisionProspect<T>>();
+  final potentialsTmp = <List<ShapeHitbox>>[];
+
+  @override
+  HashSet<CollisionProspect<T>> query() {
+    potentials.clear();
+    potentialsTmp.clear();
+
+    for (final activeItem in activeCollisions) {
+      final asShapeItem = activeItem as ShapeHitbox;
+
+      if (asShapeItem.isRemoving || asShapeItem.parent == null) {
+        tree.remove(activeItem);
+        continue;
+      }
+
+      final itemCenter = activeItem.aabb.center;
+      final markRemove = <ShapeHitbox>[];
+      final potentiallyCollide = tree.query(activeItem);
+      for (final potential in potentiallyCollide.entries.first.value) {
+        if (potential.collisionType == CollisionType.inactive) {
+          continue;
+        }
+
+        if (_broadphaseCheckCache[activeItem]?[potential] == false) {
+          continue;
+        }
+
+        final asShapePotential = potential as ShapeHitbox;
+
+        if (asShapePotential.isRemoving || asShapePotential.parent == null) {
+          markRemove.add(potential);
+          continue;
+        }
+        if (asShapePotential.parent == asShapeItem.parent &&
+            asShapeItem.parent != null) {
+          continue;
+        }
+
+        Vector2 potentialCenter;
+        if (potential.collisionType == CollisionType.passive) {
+          potentialCenter = _getCenterOfHitbox(asShapePotential);
+        } else {
+          potentialCenter = potential.aabb.center;
+        }
+
+        final distanceCloseEnough =
+            minimumDistanceCheck.call(itemCenter, potentialCenter);
+        if (distanceCloseEnough == false) {
+          continue;
+        }
+
+        potentialsTmp.add([asShapeItem, potential]);
+      }
+      markRemove.forEach(tree.remove);
+    }
+
+    if (potentialsTmp.isNotEmpty) {
+      for (var i = 0; i < potentialsTmp.length; i++) {
+        final item0 = potentialsTmp[i].first;
+        final item1 = potentialsTmp[i].last;
+        var keep = broadphaseCheck(item0, item1);
+        if (keep) {
+          keep = broadphaseCheck(item1, item0);
+        }
+        if (keep) {
+          potentials.add(CollisionProspect(item0 as T, item1 as T));
+        } else {
+          if (_broadphaseCheckCache[item0 as T] == null) {
+            _broadphaseCheckCache[item0 as T] = {};
+          }
+          _broadphaseCheckCache[item0 as T]![item1 as T] = false;
+        }
+      }
+    }
+    return potentials;
+  }
+
+  void updateItemSizeOrPosition(T item) {
```

Comment:
Does scaling or rotating a component affects scaling and rotating values of component's hitbox? 
I figured out, that changing component position does not affect to hitbox's position. 
So maybe I need to listen to component's values instead of hitbox's?

Reply:
All transforms of the parent affect the hitbox, the collision detection system is looking at the `absolutePosition` not the local position of the hitbox. The local transforms of the hitbox are for how the hitbox should be placed within the parent.